tests/test_builtin_traits.py

```diff
@@ -68,26 +68,6 @@ class HasMultipleParentOp(IRDLOperation):
     traits = frozenset([HasParent(ParentOp, Parent2Op)])
 
 
-def test_has_parent_no_parent():
-    """
-    Test that an operation with an HasParentOp trait
-    fails with no parents.
-    """
-    has_parent_op = HasParentOp()
-    with pytest.raises(
-        VerifyException, match="'test.has_parent' expects parent op 'test.parent'"
-    ):
-        has_parent_op.verify()
-
-    has_multiple_parent_op = HasMultipleParentOp()
-    message = (
-        "'test.has_multiple_parent' expects parent op to "
-        "be one of 'test.parent', 'test.parent2'"
-    )
-    with pytest.raises(VerifyException, match=message):
-        has_multiple_parent_op.verify()
-
-
```

Comment:
if we're changing the behaviour then I would expect an update of the test, to one that succeeds if the operation has no parents. (I still think a different PR would be a better place for this change)